source/cellular_bg96_api.c

```diff
@@ -2223,6 +2323,39 @@ static CellularPktStatus_t socketSendDataPrefix( void * pCallbackContext,
     return pktStatus;
 }
 
+static CellularPktStatus_t fileUploadInFileSystem( void * pCallbackContext,
+                                                 char * pLine,
+                                                 uint32_t * pBytesRead )
+{
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+
+    if( ( pLine == NULL ) || ( pBytesRead == NULL ) )
+    {
+        LogError( ( "socketSendDataPrefix: pLine is invalid or pBytesRead is invalid" ) );
+        pktStatus = CELLULAR_PKT_STATUS_BAD_PARAM;
+    }
+    else if( pCallbackContext != NULL )
+    {
+        LogError( ( "socketSendDataPrefix: pCallbackContext is not NULL" ) );
+        pktStatus = CELLULAR_PKT_STATUS_BAD_PARAM;
+    }
+    else if( *pBytesRead != 2U )

```

Comment:
Why `!=`? It might be safer and clearer to use `<`.

Also, I think the function will return OK. Shouldn't we return an error?

Reply:
Actually yes, this was not 100% correct, the pBytesRead should be !=9 and I've added a pktStatus CELLULAR_PKT_STATUS_INVALID_DATA